pkg/sql/scanner/plpgsql_scan.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package scanner
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/sql/lexbase"
+	"github.com/cockroachdb/cockroach/pkg/sql/plpgsql/parser/plpgsqllexbase"
+	"unicode/utf8"
+	"unsafe"
+)
+
+// PLPGSQLScanner is a scanner with a PLPGSQL specific scan function
+type PLPGSQLScanner struct {
```

Comment:
I think we should move this definition to a `scanner.go` within `plpgsql/parser`.